File: app/src/main/java/com/example/sendserverdemo/AccessServerManager.java
package com.example.sendserverdemo;

import retrofit2.Call;

public class AccessServerManager {
    public static volatile AccessServerManager INSTANCE;

    public AccessServerManager getInstance() {
        if (INSTANCE == null) {
            synchronized (AccessServerManager.class) {
                if (INSTANCE == null) {
                    INSTANCE = new AccessServerManager();
                }
            }
        }
        return INSTANCE;
    }

    private RetrofitService retrofitService;

    private AccessServerManager() {
        retrofitService = RetrofitManager.INSTANCE.getRetrofit().create(RetrofitService.class);
    }


    public <T> T request(BaseRequest baseRequest, BaseResponse<T> baseResponse) {

        RequestHandler requestHandler = null;
        switch (baseRequest.requestType){
            case GET:
                requestHandler = new GetRequestHandler();
                break;
            case POST:
                requestHandler = new PostRequestHandler();
                break;
        }

        requestHandler.requestHandler(retrofitService,baseRequest,baseResponse);



        return null;
    }


}

File: app/src/main/java/com/example/sendserverdemo/AccessServerManager.kt
package com.example.sendserverdemo

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class AccessServerManager private constructor() {
    private val retrofitService: RetrofitService =
        RetrofitManager.getRetrofit().create(RetrofitService::class.java)

    suspend fun <ResponseT> request(
        baseRequest: BaseRequest,
        responseTClass: Class<ResponseT>?
    ): ResponseT {
        val requestHandler: RequestHandler = when (baseRequest.requestType) {
            BaseRequest.RequestType.GET -> GetRequestHandler()
            BaseRequest.RequestType.POST -> PostRequestHandler()
        }
        return withContext(Dispatchers.IO) {
            requestHandler.requestHandler(retrofitService, baseRequest, responseTClass)
        }
    }

    companion object {
        @Volatile
        private lateinit var INSTANCE: AccessServerManager
        val instance: AccessServerManager
            get() {
                if (INSTANCE == null) {
                    synchronized(AccessServerManager::class.java) {
                        if (INSTANCE == null) {
                            INSTANCE = AccessServerManager()
                        }
                    }
                }
                return INSTANCE
            }
    }
}